Explorer/Assets/DCL/UI/MainUIContainer/MainUIController.cs

```diff
@@ -0,0 +1,156 @@
+﻿using Cysharp.Threading.Tasks;
+using DCL.Chat;
+using DCL.Minimap;
+using DCL.SidebarBus;
+using DCL.UI.Sidebar;
+using JetBrains.Annotations;
+using MVC;
+using System;
+using System.Threading;
+using UnityEngine;
+using Utility;
+
+namespace DCL.UI.MainUI
+{
+    public class MainUIController : ControllerBase<MainUIView>
+    {
+        private const float SHOW_SIDEBAR_LAYOUT_WIDTH = 80;
+        private const float HIDE_SIDEBAR_LAYOUT_WIDTH = 20;
+        private const float HIDE_SIDEBAR_WAIT_TIME = 0.3f;
+        private const float SHOW_SIDEBAR_WAIT_TIME = 0.3f;
+        private const float SIDEBAR_ANIMATION_TIME = 0.2f;
+
+        private readonly ISidebarBus sidebarBus;
+        private readonly IMVCManager mvcManager;
+
+        private bool waitingToShowSidebar;
+        private bool waitingToHideSidebar;
+        private bool showingSidebar;
+        private bool sidebarBlockStatus;
+        private bool autoHideSidebar;
+        private CancellationTokenSource showSidebarCancellationTokenSource = new ();
+        private CancellationTokenSource hideSidebarCancellationTokenSource = new ();
+
+        public override CanvasOrdering.SortingLayer Layer => CanvasOrdering.SortingLayer.Persistent;
+
+        public MainUIController(
+            [NotNull] ViewFactoryMethod viewFactory,
```

Comment:
Please use nullable annotations `ViewFactoryMethod? viewFactory`

Reply:
Ahh for some reason my Rider doesnt let me put ViewFactoryMethod! xD it automatically converts it to [NotNull]... although I see most controllers dont even have this, I think its auto-added when using Rider "implement missing members" for Interfaces...